### PR TITLE
feat(profiling): import evented profile

### PR DESCRIPTION
**Pull request by JonasBa**
Add the fuctionality of importing evented profiles. Also ported some utils that will be used in future PR's.

P.S: the PR got a bit large, but the majority are tests, so do not fear reviewing

### CODE REVIEW

File: static/app/utils/profiling/profile/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import {Frame} from 'sentry/utils/profiling/frame';
+
+export function createFrameIndex(
```
Comment (AbhiPrasad) [inline static/app/utils/profiling/profile/utils.tsx:3]: Can this just be a `array.reduce` call?

File: static/app/utils/profiling/profile/eventedProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+  leaveFrame(_event: Frame, at: number): void {
+    this.addWeightToFrames(at);
+    this.addWeightsToNodes(at);
+
+    const leavingStackTop = this.appendOrderStack.pop();
+    if (!leavingStackTop) throw new Error('Unbalanced stack');
+
+    // Lock the stack node, so we make sure we dont mutate it in the future.
+    // The samples should be ordered by timestamp when processed so we should never
```
Comment (AbhiPrasad) [inline static/app/utils/profiling/profile/eventedProfile.tsx:128]: Should we throw an error if they are not ordered by timestamp?

**Review by AbhiPrasad — COMMENTED** [review]
Will circle back if nobody else gets to this to give the ✅

File: static/app/utils/profiling/profile/eventedProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      if (!frame) {
+        throw new Error(`Cannot retrieve event: ${event.frame} from frame index`);
+      }
+
+      switch (event.type) {
+        case 'O': {
+          profile.enterFrame(frame, event.at);
+          break;
+        }
+        case 'C': {
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/utils/profiling/profile/eventedProfile.tsx:37]: @AbhiPrasad yeah, exactly. I'll add a comment, but I like having this defined on the frame type and keep it closer to the type definition of the event

File: static/app/utils/profiling/profile/eventedProfile.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/utils/profiling/profile/eventedProfile.tsx:128]: @AbhiPrasad Yeah, I dont think I ever saw this happen, but we should at least know if it does!

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**